### PR TITLE
DAOS-10346 Test: Fix the tests to use the existing container uuid instead of creating new container as part of IOR.

### PR DESCRIPTION
**Pull request by ravalsam**
- nvme fault Test was failing as it was not using the correct container uuid.
- changed enospace code to use it's own container to avoid conflict with nvme util container which runs in background
- Updated nvme utility to create it's own container instead to avoid the conflicts when tests are run in threads
- Removed the skip test as DAOS-5281 as it's not valid test case.

Test-tag: pr nvme ec

Signed-off-by: Samir Raval <samir.raval@intel.com>

### CODE REVIEW

File: src/tests/ftest/nvme/fault.yaml [hunk 1]
```diff
@@ -59,10 +59,9 @@ ior:
   iorflags:
       flags: "-w -F -r -R -k -G 1"
   test_file: /testFile
-  repetitions: 1
   transfersize_blocksize:
     16M:
-      transfer_size: 16777216 #16M
+      transfer_size: 16M
```
Comment (daltonbohning) [inline src/tests/ftest/nvme/fault.yaml:64]: This whole section could be simplified
```
  transfer_size: 16M
  dfs_oclass: "RP_2G1"
  dfs_dir_oclass: "RP_2G1"
```
Also, there isn't a `block_size` specified, so IOR will fail

**Review by daltonbohning — CHANGES_REQUESTED** [review]

File: src/tests/ftest/nvme/fault.yaml (hunk 1 above)
Comment (ravalsam) [reply src/tests/ftest/nvme/fault.yaml:64]: No test is going to calculate the block size based on how much % of pool needs to be filled up. https://github.com/daos-stack/daos/blob/643ef3af2b8ebbd0368c77bc413c0ff8106e010c/src/tests/ftest/util/nvme_utils.py#L145

Comment (daltonbohning) [reply src/tests/ftest/nvme/fault.yaml:64]: Ah, thank you :)

Comment (daltonbohning) [reply src/tests/ftest/nvme/fault.yaml:64]: So just the simplification then, if you need to push again (optional)

**Review by daltonbohning — APPROVED** [review]

**Comment by ravalsam** [comment]
Below failure test cases in Ci are because of existing defect already open or created one for new issue.

nvme/enospace.py - DAOS-10337
server/metadata.py - DAOS-9078
ior/hard_rebuild.py - DAOS-10342
erasurecode/restart.py - DAOS-10338
erasurecode/rebuild_disabled.py - DAOS-10351
erasurecode/rank_failure.py -  DAOS-10419

**Comment by ravalsam** [comment]
Below failure test cases in Ci are because of existing defect already open or created one for new issue.

nvme/enospace.py - DAOS-10337
server/metadata.py - DAOS-9078
ior/hard_rebuild.py - DAOS-10342
erasurecode/restart.py - DAOS-10338
erasurecode/rebuild_disabled.py - DAOS-10351
erasurecode/rank_failure.py -  DAOS-10419

File: src/tests/ftest/nvme/enospace.yaml [hunk 2]
```diff
@@ -60,12 +60,8 @@ ior:
...
   test_file: /testFile
-  repetitions: 1
   transfersize_blocksize:
     2K:
       transfer_size: 2048 #2K
     16M:
       nvme_transfer_size: 16777216 #16M
-  objectclass:
-    S1:
-      obj_class: "S1"
```
Comment (daltonbohning) [inline src/tests/ftest/nvme/enospace.yaml:71]: Should this specify `dfs_oclass`?

File: src/tests/ftest/util/nvme_utils.py [hunk 3]
```diff
@@ -92,6 +95,18 @@ def setUp(self):
...
+        """Create the container """
+        # Get container params
+        self.nvme_local_cont = TestContainer(self.pool, daos_command=DaosCommand(self.bin))
+        self.nvme_local_cont.get_params(self)
+
+        # update container oclass
+        if self.ior_local_cmd.dfs_oclass:
+            self.nvme_local_cont.oclass.update(self.ior_local_cmd.dfs_oclass.value)
+
+        self.nvme_local_cont.create()
```
Comment (daltonbohning) [inline src/tests/ftest/util/nvme_utils.py:108]: I'd recommend using this existing function:
https://github.com/daos-stack/daos/blob/2b0d4a638977a4f8c7a0566ea9a8160ed0bcd573/src/tests/ftest/util/apricot/apricot/test.py#L1669

```suggestion
        # Get container params
        self.nvme_local_cont = self.get_container(self.pool, create=False)

        # update container oclass
        if self.ior_local_cmd.dfs_oclass:
            self.nvme_local_cont.oclass.update(self.ior_local_cmd.dfs_oclass.value)

        self.nvme_local_cont.create()
```

File: src/tests/ftest/nvme/enospace.py [hunk 4]
```diff
@@ -100,8 +101,13 @@ def ior_bg_thread(self):
 
         # Define the job manager for the IOR command
         job_manager = get_job_manager(self, "Mpirun", ior_bg_cmd, mpi_type="mpich")
-        self.create_cont()
-        job_manager.job.dfs_cont.update(self.container.uuid)
+
+        # create container
+        container = TestContainer(self.pool, daos_command=DaosCommand(self.bin))
+        container.get_params(self)
+        container.create()
```
Comment (daltonbohning) [inline src/tests/ftest/nvme/enospace.py:108]: I'd recommend using this existing function:
https://github.com/daos-stack/daos/blob/2b0d4a638977a4f8c7a0566ea9a8160ed0bcd573/src/tests/ftest/util/apricot/apricot/test.py#L1669
```suggestion
        # create container
        container = self.get_container(self.pool)
```

**Review by daltonbohning — DISMISSED** [review]
LGTM, but could we please use the existing `get_container` method as possible?

File: src/tests/ftest/nvme/enospace.yaml (hunk 2 above)
Comment (ravalsam) [reply src/tests/ftest/nvme/enospace.yaml:71]: Not needed

**Review by daltonbohning — DISMISSED** [review]
LGTM, but recommend using the existing `get_container` if you have to push again

**Comment by ravalsam** [comment]
> LGTM, but recommend using the existing `get_container` if you have to push again

Yes somehow batch push overwritten when I rebase. I updated again.

File: src/tests/ftest/nvme/enospace.py [hunk 5]
```diff
@@ -15,6 +15,7 @@
 from ior_utils import IorCommand, IorMetrics
 from exception_utils import CommandFailure
 from general_utils import error_count
+from test_utils_container import TestContainer
```
Comment (daltonbohning) [inline src/tests/ftest/nvme/enospace.py:18]: We should remove this

File: src/tests/ftest/util/nvme_utils.py [hunk 6]
```diff
@@ -13,6 +13,8 @@
 from ior_utils import IorCommand
 from server_utils import ServerFailed
 from job_manager_utils import get_job_manager
+from daos_utils import DaosCommand
+from test_utils_container import TestContainer
```
Comment (daltonbohning) [inline src/tests/ftest/util/nvme_utils.py:17]: We should remove this since using `get_container`

File: src/tests/ftest/util/ec_utils.py [hunk 7]
```diff
@@ -67,6 +67,7 @@ def check_aggregation_status(pool, quick_check=True, attempt=20):
 
 class ErasureCodeIor(ServerFillUp):
     # pylint: disable=too-many-ancestors
+    # pylint: disable=attribute-defined-outside-init
```
Comment (daltonbohning) [inline src/tests/ftest/util/ec_utils.py:70]: I think we should instead just set `self.nvme_local_cont = None` in `__init__`

**Review by daltonbohning — APPROVED** [review]

**Review by rpadma2 — CHANGES_REQUESTED** [review]
LGTM.

**Review by rpadma2 — APPROVED** [review]

**Pull request merged**